src/main/java/com/labsynch/labseer/api/ApiSaltController.java

```diff
@@ -82,6 +95,191 @@ public ResponseEntity<String> showJson(@PathVariable("id") Long id) {
+		headers.add("Access-Control-Allow-Origin", "*");
+		headers.add("Access-Control-Allow-Headers", "Content-Type");
+		headers.add("Cache-Control", "no-store, no-cache, must-revalidate"); // HTTP 1.1
+		headers.add("Pragma", "no-cache"); // HTTP 1.0
+		headers.setExpires(0); // Expire the cache
+		if (oldSalt == null) {
+			return new ResponseEntity<String>("Salt could not be found", headers, HttpStatus.NOT_FOUND);
+		}
+		else
+		{
+			try {
+				ObjectMapper objectMapper = new ObjectMapper();
+
+				JsonNode rootNode = objectMapper.readTree(json);
+				JsonNode dryRunNode = rootNode.path("dryrun");
+				boolean dryrun = dryRunNode.asBoolean();
+
+				rootNode = objectMapper.readTree(json);
+				JsonNode saltNode = rootNode.path("saltJSON");
+				Salt newSalt = Salt.fromJsonToSalt(saltNode.toString());
+
+				ArrayList<ErrorMessage> warnings = new ArrayList<ErrorMessage>();
+
+				if(!newSalt.getAbbrev().equals(oldSalt.getAbbrev())){
+					ErrorMessage warning = new ErrorMessage();
+					warning.setLevel("warning");
+					warning.setMessage("The abbreviation of this salt will be changed.");
+					warnings.add(warning);
+				}
+				if(!newSalt.getName().equals(oldSalt.getName())){
+					ErrorMessage warning = new ErrorMessage();
+					warning.setLevel("warning");
+					warning.setMessage("The name of this salt will be changed.");
+					warnings.add(warning);
+				}
+
+				newSalt.setAbbrev(newSalt.getAbbrev().trim());
+				newSalt.setName(newSalt.getName().trim());
+				newSalt.setFormula(saltStructureService.calculateFormula(newSalt));
+
+				// This doesn't always work; Might need to convert to mol and then compare
+				if(!newSalt.getMolStructure().equals(oldSalt.getMolStructure())){
+					ErrorMessage warning = new ErrorMessage();
+					warning.setLevel("warning");
+					warning.setMessage("The structure of this salt might be changed. The weight and charge might be affected.");
+					warnings.add(warning);
+				}
+
+				boolean validSalt = true;
+				List<Salt> saltsByName = Salt.findSaltsByNameEquals(newSalt.getName()).getResultList();
+				if (saltsByName.size() > 1) {
+					logger.error("Number of salts found: " + saltsByName.size());
+					validSalt = false;
+					ErrorMessage error = new ErrorMessage();
+					error.setLevel("error");
+					error.setMessage("Duplicate salt name. Another salt exist with the same name.");
+					warnings.add(error);
+				}
+				List<Salt> saltsByAbbrev = Salt.findSaltsByAbbrevEquals(newSalt.getAbbrev()).getResultList();
+				if (saltsByAbbrev.size() > 1) {
+					logger.error("Number of salts found: " + saltsByAbbrev.size());
+					validSalt = false;
+					ErrorMessage error = new ErrorMessage();
+					error.setLevel("error");
+					error.setMessage("Duplicate salt abbreviation. Another salt exist with the same abbreviation.");
+					warnings.add(error);
+				}
+				List<Salt> saltsByFormula = Salt.findSaltsByFormulaEquals(newSalt.getFormula()).getResultList();
```

Comment:
Now that I look at this again there is such a thing as isomers so I will want to change this to checking duplicate structures

Reply:
maybe I will keep this but set the level to warning

Reply:
I like the idea of keeping it but with a warning.